packages/@n8n/nodes-langchain/nodes/llms/LmChatGoogleVertex/LmChatGoogleVertex.node.ts

```diff
@@ -0,0 +1,250 @@
+					{
+						displayName: 'ID',
+						name: 'id',
+						type: 'string',
+					},
+				],
+			},
+			{
+				displayName: 'Model Name',
+				name: 'modelName',
+				type: 'string',
+				description:
+					'The model which will generate the completion. <a href="https://cloud.google.com/vertex-ai/generative-ai/docs/learn/models">Learn more</a>.',
+				default: 'gemini-1.5-flash',
+			},
+			{
+				displayName: 'Options',
+				name: 'options',
+				placeholder: 'Add Option',
+				description: 'Additional options to add',
+				type: 'collection',
+				default: {},
+				options: [
+					{
+						displayName: 'Maximum Number of Tokens',
+						name: 'maxOutputTokens',
+						default: 2048,
+						description: 'The maximum number of tokens to generate in the completion',
+						type: 'number',
+					},
+					{
+						displayName: 'Sampling Temperature',
+						name: 'temperature',
+						default: 0.4,
+						typeOptions: { maxValue: 1, minValue: 0, numberPrecision: 1 },
+						description:
+							'Controls randomness: Lowering results in less random completions. As the temperature approaches zero, the model will become deterministic and repetitive.',
+						type: 'number',
+					},
+					{
+						displayName: 'Top K',
+						name: 'topK',
+						default: 32,
+						typeOptions: { maxValue: 40, minValue: -1, numberPrecision: 1 },
+						description:
+							'Used to remove "long tail" low probability responses. Defaults to -1, which disables it.',
+						type: 'number',
+					},
+					{
+						displayName: 'Top P',
+						name: 'topP',
+						default: 1,
+						typeOptions: { maxValue: 1, minValue: 0, numberPrecision: 1 },
+						description:
+							'Controls diversity via nucleus sampling: 0.5 means half of all likelihood-weighted options are considered. We generally recommend altering this or temperature but not both.',
+						type: 'number',
+					},
+
+					// Safety Settings
+					{
+						displayName: 'Safety Settings',
+						name: 'safetySettings',
+						type: 'fixedCollection',
+						typeOptions: { multipleValues: true },
+						default: {
+							values: {
+								category: harmCategories[0].name as HarmCategory,
+								threshold: harmThresholds[0].name as HarmBlockThreshold,
+							},
+						},
+						placeholder: 'Add Option',
+						options: [
+							{
+								name: 'values',
+								displayName: 'Values',
+								values: [
+									{
+										displayName: 'Safety Category',
+										name: 'category',
+										type: 'options',
+										description: 'The category of harmful content to block',
+										default: 'HARM_CATEGORY_UNSPECIFIED',
+										options: harmCategories,
+									},
+									{
+										displayName: 'Safety Threshold',
+										name: 'threshold',
+										type: 'options',
+										description: 'The threshold of harmful content to block',
+										default: 'HARM_BLOCK_THRESHOLD_UNSPECIFIED',
+										options: harmThresholds,
+									},
+								],
+							},
+						],
+					},
```

Comment:
These are identical to the options from ChatGoogleGemini, right? Would be nice to move them to some shared module and import both here and in ChatGoogleGemini. Same for `harmCategories` and `harmThresholds`

Reply:
Moved them both to gemini-common folder, is it ok?